lib/net.c

```diff
@@ -42,17 +44,45 @@
 
 static struct net_config config;
 
-/*
- * XXX The secret key of the RSS hash must be random
- * in order to avoid hackers to know it.
- */
-uint8_t default_rss_key[GATEKEEPER_RSS_KEY_LEN] = {
-	0x6d, 0x5a, 0x56, 0xda, 0x25, 0x5b, 0x0e, 0xc2,
-	0x41, 0x67, 0x25, 0x3d, 0x43, 0xa3, 0x8f, 0xb0,
-	0xd0, 0xca, 0x2b, 0xcb, 0xae, 0x7b, 0x30, 0xb4,
-	0x77, 0xcb, 0x2d, 0xa3, 0x80, 0x30, 0xf2, 0x0c,
-	0x6a, 0x42, 0xb7, 0x3b, 0xbe, 0xac, 0x01, 0xfa,
-};
+uint8_t default_rss_key[GATEKEEPER_RSS_KEY_LEN];
+
+static void
+randomize_rss_key(void)
+{
+	uint8_t i;
+	uint16_t final_set_count = 0;
+	ssize_t number_of_bytes;
+
+	/*
+	 * The secret key of the RSS hash (RSK) must be random in order
+	 * to prevent hackers from knowing it.
+	 * To validate if the key generated is reasonable, the
+	 * number of bits set to 1 in the key must be greater than 
+	 * 10% and less than 90% of the total bits in the key.
+	 * min_num_set_bits and max_num_set_bits represent the lower 
+	 * and upper bound for the key.
+	 */
+	uint16_t min_num_set_bits = sizeof(default_rss_key) * 8 * 0.1;
+	uint16_t max_num_set_bits = sizeof(default_rss_key) * 8 * 0.9;
+
+	while (final_set_count < min_num_set_bits ||
+			final_set_count > max_num_set_bits) {
```

Comment:
This loop should be a `do..while` since there's no reason to test before going through the loop. This will allow us to drop the initialization of variable `final_set_count` at the opening of the function.